spec/models/take_medication_reminder_spec.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: take_medication_reminders
+#
+#  id            :integer          not null, primary key
+#  medication_id :integer          not null
+#  active        :boolean          not null
+#  created_at    :datetime
+#  updated_at    :datetime
+#
+
+describe TakeMedicationReminder do
+  let(:user) { FactoryGirl.create(:user1) }
+  let(:medication) { FactoryGirl.create(:medication, userid: user.id) }
+  let(:reminder) { 
+    FactoryGirl.create(:take_medication_reminder, medication_id: medication.id) 
+  }
+
+  describe '#active_reminders' do                  
+    it 'returns only active reminders' do
+      expect(TakeMedicationReminder.active).to eq([reminder])
+    end        
+  end
+  
+  describe 'for_day' do    
+    let!(:weekly_medication) { 
+      FactoryGirl.create(:medication, userid: user.id, weekly_dosage: [0,2,4,6]) 
+    }
+    let!(:weekly_medication2) { 
+      FactoryGirl.create(:medication, userid: user.id, weekly_dosage: [1,2,4,6]) 
+    }
+    
+    it 'returns only sundays reminder' do
```

Comment:
Could you explain why Sunday's reminder is only returned? Sorry, I don't quite follow why that's happening.

Reply:
I am trying to test the scope `TakeMedicationReminder.for_day` which returns the available medication on the given day.
I added two medications one which has dosage on Sunday (and also other days) and other one without dosage on Sunday.
Here I am trying to query the TakeMedicationReminder for Sunday.

I have updated the messages in the test, hope its more clear now.